### PR TITLE
Check for all (potentially unsafe) HTTP methods

### PR DESCRIPTION
**Pull request by rain0r**
HTTP_METHOD_TRACE
    HTTP_METHOD_TRACK
    HTTP_METHOD_CONNECT
    HTTP_METHOD_PUT
    HTTP_METHOD_DELETE
and raise an alert if the server responds to those methods

### CODE REVIEW

File: src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java [hunk 1]
```diff
@@ -136,320 +168,428 @@ public String getReference() {
...
+		} else if (httpMethod.equals("TRACE")) {
+			ret = new HttpTrace(urlStr);
+		} else if (httpMethod.equals("OPTIONS")) {
+			ret = new HttpOptions(urlStr);
+		}
+
+		return ret;
+	}
+
+	private int tryHttpMethod(HttpRequestBase method) throws Exception {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java]: It should use the method `sendAndReceive` to send the messages, as is now it bypasses the core configurations (e.g. auth, proxy, mode...).

**Comment by kingthorin** [comment]
@rain0r closed?

**Comment by rain0r** [comment]
Yes... I'm re-working it and will re-submit a PR which will be easier to review.

**Comment by kingthorin** [comment]
Great, was concerned that you were abandoning. If you have any questions or need advice just hit us up on the dev group:
https://groups.google.com/forum/m/#!forum/zaproxy-develop

**Comment by rain0r** [comment]
See https://github.com/zaproxy/zap-extensions/pull/837

**Pull request closed without merging**